### ISSUE TITLE
Approval process fails when a scope is selected during the approval.

### ISSUE THREAD

**Issue by pankaj-cgi** [bcgov/api-services-portal#136]
# API Services Portal Issue
Approval process fails when a scope is selected during the approval.

My issuer configuration:
kind: CredentialIssuer
name: Resource Server hns-esb-dev
namespace: hns-esb-dev
description: Authorization Profile for protecting Ministry of health
flow: client-credentials
mode: auto
authPlugin: jwt-keycloak
clientAuthenticator: client-secret
clientRoles: []
availableScopes: [system/*.write]
owner: pkathuri@idir
environmentDetails:
  - environment: dev
    issuerUrl: https://common-logon-dev.hlth.gov.bc.ca/auth/realms/v2_pos
    clientId: api_gateway_client_manager
    clientRegistration: managed
    clientSecret: "<secret>"

When an admin is approving the API request, this option in Permissions tab: 
![approval-request-failed](https://user-images.githubusercontent.com/79924447/127045508-f2420e9d-152e-4c11-9321-70360327158c.JPG)

When 'Checkbox' in Roles is selected: Request Approved
When 'Checkbox' in Scopes is selected: Approval Failed

### ENV

- [ ] `DEV`
- [x ] `TEST` 
- [ ] `PROD`

### TESTCASE

-

### EXPECTED

Approval to pass when Scope is selected.

-

### ACTUAL

Approval fails when Scope is selected.

**Comment by ikethecoder** [comment]
The error showing in the Activity (Request failed with status code 500) is not very useful as the root cause is:
```
{"error":"insufficient_scope","error_description":"Policy 'Allowed Client Scopes' rejected request to client-registration service. Details: Not permitted to use specified clientScope"}
```

> The 500 appears to be due to a failed reduce() when handling the original 403 error:

(error.errors || []).reduce is not a function","stack":"TypeError: (error.errors || []).reduce is not a function\n    at flattenNestedErrors (/app/node_modules/@keystonejs/keystone/lib/Keystone/format-error.js:54:24)\n    at formatError (/app/node_modules/@keystonejs/keystone/lib/Keystone/format-error.js:98:30)\n    at /app/node_modules/@keystonejs/keystone/node_modules/apollo-server-errors/dist/index.js:168:34\n    at Array.map (<anonymous>)\n    at Object.formatApolloErrors (/app/node_modules/@keystonejs/keystone/node_modules/apollo-server-errors/dist/index.js:166:27)\n    at formatErrors (/app/node_modules/@keystonejs/keystone/node_modules/apollo-server-core/dist/requestPipeline.js:306:43)\n    at Object.<anonymous> (/app/node_modules/@keystonejs/keystone/node_modules/apollo-server-core/dist/requestPipeline.js:192:98)\n    at Generator.next (<anonymous>)\n   ...

**Comment by ikethecoder** [comment]
Root cause of this was some unnecessary scopes included during Client Registration.  This was fixed in v1.0.56.

There is still some investigation required to see why the 500 error happens.

**Comment by ikethecoder** [comment]
PR #418 improves the error handling so that the message logged to the Activity is more informative and the backend logging is improved.
